include/zephyr/bluetooth/classic/avdtp.h

```diff
@@ -4,49 +4,154 @@
+	 * signal. This also includes the situation when an INT receives a request for the
+	 * same command that it is currently expecting a response
+	 */
+	BT_AVDTP_BAD_STATE = 0x31,
+};
+
+/** @brief Stream End Point Role */
+enum bt_avdtp_role_type {
+	/** Source Role */
+	BT_AVDTP_SOURCE = 0,
+	/** Sink Role */
+	BT_AVDTP_SINK = 1
+};
+
+/** @brief Stream End Point Media Type */
+enum bt_avdtp_media_type {
+	/** Audio Media Type */
+	BT_AVDTP_AUDIO = 0x00,
+	/** Video Media Type */
+	BT_AVDTP_VIDEO = 0x01,
+	/** Multimedia Media Type */
+	BT_AVDTP_MULTIMEDIA = 0x02
+};
+
+/** @brief AVDTP SEID Information AVDTP_SPEC V13 Table 8.8 */
 struct bt_avdtp_seid_info {
-	/** Stream End Point ID */
+#ifdef CONFIG_LITTLE_ENDIAN
+	/** Reserved */
+	uint8_t rfa0:1;
+	/** End Point usage status */
+	uint8_t inuse:1;
+	/** Stream End Point ID that is the identifier of the stream endpoint */
+	uint8_t id:6;
+	/** Reserved */
+	uint8_t rfa1:3;
+	/** This field indicates if the stream end-point is SNK or SRC, @ref bt_avdtp_role_type */
+	uint8_t tsep:1;
+	/** Media-type of the End Point, @ref bt_avdtp_media_type
+	 * Only @ref BT_AVDTP_AUDIO is supported now.
+	 */
+	uint8_t media_type:4;
+#else
+	/** Stream End Point ID that is the identifier of the stream endpoint */
 	uint8_t id:6;
 	/** End Point usage status */
 	uint8_t inuse:1;
 	/** Reserved */
 	uint8_t rfa0:1;
-	/** Media-type of the End Point */
+	/** Media-type of the End Point, @ref bt_avdtp_media_type
+	 * Only @ref BT_AVDTP_AUDIO is supported now.
+	 */
 	uint8_t media_type:4;
-	/** TSEP of the End Point */
+	/** This field indicates if the stream end-point is SNK or SRC, @ref bt_avdtp_role_type */
 	uint8_t tsep:1;
 	/** Reserved */
 	uint8_t rfa1:3;
+#endif
 } __packed;
 
-/** @brief AVDTP Local SEP*/
-struct bt_avdtp_seid_lsep {
-	/** Stream End Point information */
-	struct bt_avdtp_seid_info sep;
-	/** Pointer to next local Stream End Point structure */
-	struct bt_avdtp_seid_lsep *next;
+/** @brief service category Type */
+enum bt_avdtp_service_category {
+	/** Media Transport */
+	BT_AVDTP_SERVICE_MEDIA_TRANSPORT = 0x01,
+	/** Reporting */
+	BT_AVDTP_SERVICE_REPORTING = 0x02,
+	/** Recovery */
+	BT_AVDTP_SERVICE_MEDIA_RECOVERY = 0x03,
+	/** Content Protection */
+	BT_AVDTP_SERVICE_CONTENT_PROTECTION = 0x04,
+	/** Header Compression */
+	BT_AVDTP_SERVICE_HEADER_COMPRESSION = 0x05,
+	/** Multiplexing */
+	BT_AVDTP_SERVICE_MULTIPLEXING = 0x06,
+	/** Media Codec */
+	BT_AVDTP_SERVICE_MEDIA_CODEC = 0x07,
+	/** Delay Reporting */
+	BT_AVDTP_SERVICE_DELAY_REPORTING = 0x08,
 };
 
-/** @brief AVDTP Stream */
-struct bt_avdtp_stream {
-	struct bt_l2cap_br_chan chan; /* Transport Channel*/
-	struct bt_avdtp_seid_info lsep; /* Configured Local SEP */
-	struct bt_avdtp_seid_info rsep; /* Configured Remote SEP*/
-	uint8_t state; /* current state of the stream */
-	struct bt_avdtp_stream *next;
+/** @brief AVDTP Stream End Point Identifier */
+struct bt_avdtp_seid {
```

Comment:
I'm not sure you have the struct naming correct here. The SEID is just a numeric value that refers to a SEP, so it seems counterintuitive to me to have a struct (which contains much more than the numeric id) called `bt_avdtp_seid`. Wouldn't `struct bt_avdt_sep` be a more appropriate name?

Reply:
Updated